src/corecel/sys/Device.hh

```diff
@@ -76,6 +76,15 @@ class Device
     // Construct from device ID
     explicit Device(int id);
 
+    //// DESTRUCTORS ////
+
+    // Needed for StreamStorage pimpl
+    ~Device();
+
+    //// OPERATORS ////
+
+    Device& operator=(Device&&);
```

Comment:
You're fond of pointing out the rule of 5 😉

The better way to handle this, by the way, is to define a custom deleter for stream storage, declare it in this header, and define it in the Device.cc file. Then the default constructor and move constructors are handled natively by the unique pointer.

Reply:
we end up having to use new/delete with the custom Deleter though instead of the compiler doing that for us, I'm not convinced I like that more.

Reply:
Yeah, not being able to use `make_unique` is the downside to this approach.